### PR TITLE
chore: clippy fixes

### PR DESCRIPTION
**Pull request by hrls**
#### Problem
Clippy warns `sort` on primitives.
#### Summary of Changes
A few oneliners.

### CODE REVIEW

File: sdk/src/hard_forks.rs [hunk 1]
```diff
@@ -22,7 +22,7 @@ impl HardForks {
         } else {
             self.hard_forks.push((new_slot, 1));
         }
-        self.hard_forks.sort();
+        self.hard_forks.sort_unstable();
```
Comment (mvines) [inline sdk/src/hard_forks.rs:25]: Can you remove all the `sort_unstable()` changes please.  Unstable sorts can break consensus, so we'll need to carefully consider each sort -> sort_unstable change individually

Comment (hrls) [reply sdk/src/hard_forks.rs:25]: All replaces of `sort` to `sort_unstable` in this PR targets exactly `Vec` of `Slot`. `Slot` is `Copy` type. And for `Vec<T: Copy>` there is no difference between `sort` and `sort_unstable`.

Comment (hrls) [reply sdk/src/hard_forks.rs:25]: I can revert all `sort_unstable()` back to `sort()` and marks them as `#[allow(clippy::stable_sort_primitive)]`.

Comment (mvines) [reply sdk/src/hard_forks.rs:25]: My preference is to do that.   I'm worried also about a future refactoring that might change result in type change where `sort_unstable()` is no longer equivalent to `sort()`, and accidentally breaking consensus.  `clippy::stable_sort_primitive` is borderline harmful IMO, the implementation of `sort()` should just be improved for the types were `sort_unstable()` and `sort()` are equivalent.

Comment (hrls) [reply sdk/src/hard_forks.rs]: Agree. Fixed.

**Comment by mvines** [comment]
Looks like some build failures.  You should be able to reproduce by running `ci/test-checks.sh`

**Comment by hrls** [comment]
refreshed

**Comment by mvines** [comment]
@hrls - hey do you know when `#[allow(clippy::rc_buffer)]` was introduced, Rust 1.47?

**Comment by hrls** [comment]
@mvines hi, looks like 1.47 https://github.com/rust-lang/rust-clippy/pull/6128

**Comment by mvines** [comment]
Ok darn.  We're stuck on 1.46 right now.  Would you mind removing those `#![allow(clippy::rc_buffer)]` for now, so we can get the rest of this PR landed

**Comment by quasiyoke** [comment]
@hrls Nice PR BTW. Please check your email, we can't connect for several months. Sorry for offtop, guys.

**Comment by mvines** [comment]
Thanks for the work here @hrls! I've rolled it into https://github.com/solana-labs/solana/pull/14100, it certainly saved me some time

**Pull request closed without merging**